native/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java

```diff
@@ -0,0 +1,629 @@
+                                }
+
+                            }
+                            entityArray.add(index, new Entity(getArray(keys), tableName, module));
+
+                        }
+
+                    }
+                }
+            }
+            RecordTypeDescriptorNode recordDesc = (RecordTypeDescriptorNode) ((TypeDefinitionNode) moduleNode)
+                    .typeDescriptor();
+            entityArray.get(index).setEntityName(((TypeDefinitionNode) moduleNode).typeName().text());
+            for (Node node : recordDesc.fields()) {
+                if (node.kind() == SyntaxKind.RECORD_FIELD_WITH_DEFAULT_VALUE) {
+                    RecordFieldWithDefaultValueNode fieldNode = (RecordFieldWithDefaultValueNode) node;
+                    String fName = fieldNode.fieldName().text();
+                    String fType = fieldNode.typeName().toString();
+                    HashMap<String, String> map = new HashMap<>();
+                    if (((RecordFieldWithDefaultValueNode) node).metadata().isEmpty()) {
+                        map.put(BalFileConstants.FIELD_NAME, fName);
+                        map.put(BalFileConstants.FIELD_TYPE, fType);
+                        map.put(BalFileConstants.AUTOGENERATED, BalFileConstants.FALSE);
+                    } else {
+                        MetadataNode fieldMetaD = ((RecordFieldWithDefaultValueNode) node).metadata().get();
+                        map.put(BalFileConstants.FIELD_NAME, fName);
+                        map.put(BalFileConstants.FIELD_TYPE, fType);
+                        map.put(BalFileConstants.AUTOGENERATED, readMetaData(fieldMetaD));
+                    }
+                    entityArray.get(index).addField(map);
+                } else if (node.kind() == SyntaxKind.RECORD_FIELD) {
+                    RecordFieldNode fieldNode = (RecordFieldNode) node;
+                    String fName = fieldNode.fieldName().text();
+                    String fType = fieldNode.typeName().toString();
+                    HashMap<String, String> map = new HashMap<>();
+                    if (((RecordFieldNode) node).metadata().isEmpty()) {
+                        map.put(BalFileConstants.FIELD_NAME, fName);
+                        map.put(BalFileConstants.FIELD_TYPE, fType);
+                        map.put(BalFileConstants.AUTOGENERATED, BalFileConstants.FALSE);
+
+                    } else {
+                        MetadataNode fieldMetaD = ((RecordFieldNode) node).metadata().get();
+                        map.put(BalFileConstants.FIELD_NAME, fName);
+                        map.put(BalFileConstants.FIELD_TYPE, fType);
+                        map.put(BalFileConstants.AUTOGENERATED, readMetaData(fieldMetaD));
+
+                    }
+                    entityArray.get(index).addField(map);
+                }
+            }
+        }
+        return entityArray;
+    }
+
+    public static SyntaxTree generateBalFile(Entity entity) {
```

Comment:
shall we rename the function to `generateSyntaxTree`

Reply:
Updated in [d6a53d3](https://github.com/ballerina-platform/persist-tools/pull/15/commits/d6a53d3f948c8f076acb3cc1f2730258a998cf40)